### PR TITLE
chore: regenerate conversion code

### PR DESCRIPTION
**Pull request by sdowell**
The conversion code for HelmBase is lossy and therefore must be done manually. This change re-runs the automated conversion code generation and adds the manual conversion function.

Adds a dependency to the clientgen target to make it easier to run in cases where the module is not in the canonical gopath.

### CODE REVIEW

File: pkg/api/configsync/v1alpha1/conversion.go [hunk 1]
```diff
@@ -31,3 +31,11 @@ import (
 func Convert_v1beta1_HelmRootSync_To_v1alpha1_HelmRootSync(in *v1beta1.HelmRootSync, out *HelmRootSync, s conversion.Scope) error {
 	return autoConvert_v1beta1_HelmRootSync_To_v1alpha1_HelmRootSync(in, out, s)
 }
+
+// Convert_v1beta1_HelmBase_To_v1alpha1_HelmBase converts HelmBase from v1beta1
+// to v1alpha1.
+//
```
Comment (karlkfi) [inline pkg/api/configsync/v1alpha1/conversion.go:37]: add a comment explaining why this is manual and how it's lossy

**Review by karlkfi — CHANGES_REQUESTED** [review]

File: Makefile.gen [hunk 2]
```diff
@@ -16,8 +16,16 @@
 
 ######################################################################
 
+CANONICAL_GOPATH := $(shell go env GOPATH)/src/kpt.dev/configsync
+
+.PHONY: symlink-gopath
+symlink-gopath:
+	test -d $(CANONICAL_GOPATH) || ln -s $(shell pwd) $(CANONICAL_GOPATH)
```
[the remark this answers is not in the document]
Comment (sdowell) [reply Makefile.gen:23]: No it doesn't, but I figure we shouldn't mess with the users filesystem if they already have something at that path

File: pkg/api/configsync/v1alpha1/conversion.go (hunk 1 above)
Comment (sdowell) [reply pkg/api/configsync/v1alpha1/conversion.go:37]: Do we want to copy/paste the same comment for every function signature? Seems redundant

File: Makefile.gen (hunk 2 above)
[the remark this answers is not in the document]
Comment (sdowell) [reply Makefile.gen]: Added a check that causes error if the path is different

File: pkg/api/configsync/v1alpha1/conversion.go (hunk 1 above)
Comment (sdowell) [reply pkg/api/configsync/v1alpha1/conversion.go:37]: Added a comment

File: Makefile.gen [hunk 3]
```diff
@@ -16,8 +16,17 @@
 
 ######################################################################
 
+CANONICAL_GOPATH := $(shell go env GOPATH)/src/kpt.dev/configsync
+
+.PHONY: symlink-gopath
+symlink-gopath:
+	@[[ -d $(CANONICAL_GOPATH) ]] || ln -s $(shell pwd) $(CANONICAL_GOPATH)
+	@[[ $(shell pwd) == $(shell realpath $(CANONICAL_GOPATH)) ]]
```
Comment (karlkfi) [inline Makefile.gen:24]: fwiw, i dont think realpath exists on mac by default, but it can be installed.

Comment (karlkfi) [reply Makefile.gen:24]: would this exit if not true? or does it just continue?

Comment (sdowell) [reply Makefile.gen:24]: The statement returns a nonzero exit code, causing the make target to error

Comment (karlkfi) [reply Makefile.gen:24]: can you make it print an error message on failure? to stderr perhaps

Comment (sdowell) [reply Makefile.gen]: done

**Review by karlkfi — APPROVED** [review]
/lgtm

**Pull request merged**